src/emmental/model.py

```diff
@@ -256,6 +256,38 @@ def __repr__(self) -> str:
         cls_name = type(self).__name__
         return f"{cls_name}(name={self.name})"
 
+    def _get_data_from_output_dict(
```

Comment:
This function is still a bit confusing. The comments don't seem to match the code exactly (e.g., the first one is looking at index of str or int but comment is str).

I think what would help is a more detailed comment on top explaining the high level logic of what you consider or not as valid inputs.  I think you have a similar comment in the changelog.

Also, what is the very last elif handling where int(index[1]) == 0?

Reply:
1. Thanks! Updated the comments.
2. Update the design idea in the docstring.
3. Great catch! Simplified the code. It tries to handle that there is only one output of module and and the action wants to have that output with index 0.